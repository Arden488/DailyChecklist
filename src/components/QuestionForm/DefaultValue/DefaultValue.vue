<template>
  <div>
    <el-input-number 
      v-if="type === 'mood' || type ==='time'"
      v-model="valueLocal" 
      :min="0" 
      :max="+options.rangeEnd"
      @change="updateValue($event)"
    ></el-input-number>

    <el-switch v-if="type === 'boolean'" v-model="valueLocal" @change="updateValue($event)"></el-switch>

    <el-select 
      v-if="type === 'select'" 
      v-model="valueLocal"
      @change="updateValue($event)"
      placeholder="Choose default"
    >
      <el-option
        v-for="option in options.values"
        :key="option.key"
        :value="option.label"
      ></el-option>
    </el-select>
  </div>
</template>

<script>
export default {
  name: 'DefaultValue',
  props: [
    'type',
    'options',
    'value'
  ],
  data () {
    return {
      valueLocal: this.value,
    }
  },
  methods: {
    updateValue (value) {
      this.$emit('defaultValue', value);
    }
  }
}
</script>

