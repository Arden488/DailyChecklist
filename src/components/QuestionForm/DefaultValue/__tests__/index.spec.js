/* global it, describe, expect */

import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vue from 'vue';
import VueRouter from 'vue-router';
import ElementUI from 'element-ui';
import DefaultValue from '../DefaultValue.vue';

const localVue = createLocalVue();
localVue.use(VueRouter);

describe('DefaultValue layout', () => {
  it('is a Vue instance', () => {
    localVue.use(ElementUI);
    const wrapper = shallowMount(DefaultValue, { 
      localVue
    });
    expect(wrapper.isVueInstance).toBeTruthy()
  })

  it('is renders correctly', () => {
    const wrapper = shallowMount(DefaultValue, { 
      localVue
    });
    expect(wrapper).toMatchSnapshot()
  })
});
