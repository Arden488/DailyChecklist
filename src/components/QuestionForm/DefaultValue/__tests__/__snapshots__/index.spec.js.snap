// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultValue layout is renders correctly 1`] = `
<div>
  <!---->
  <!---->
  <!---->
</div>
`;
